runtime/parachains/src/paras.rs

```diff
@@ -536,6 +536,37 @@ impl<T: Trait> Module<T> {
 
 		Self::past_code_meta(&id).most_recent_change()
 	}
+
+	/// Compute the local-validation data based on the head of the para. This assumes the
+	/// relay-parent is the parent of the current block.
+	pub(crate) fn local_validation_data(para_id: ParaId) -> Option<LocalValidationData<T::BlockNumber>> {
+		let relay_parent_number = <system::Module<T>>::block_number() - One::one();
+
+		let config = <configuration::Module<T>>::config();
+		let freq = config.validation_upgrade_frequency;
+		let delay = config.validation_upgrade_delay;
+
+		let last_code_upgrade = Self::last_code_upgrade(para_id, true);
+		let can_upgrade_code = last_code_upgrade.map_or(
+			true,
+			|l| l <= relay_parent_number && relay_parent_number.saturating_sub(l) >= freq,
```

Comment:
```suggestion
			|l| { l <= relay_parent_number && relay_parent_number.saturating_sub(l) >= freq },
```
up for debate, but I find it harder to read without the explicit `{` scope `}`